gnn/plot.py

```diff
@@ -99,41 +183,120 @@ def plot_top_models_and_baselines(
     # plot the top N GNNs performance
     for model, metrics in top_model_2_metrics.items():
         ax.bar(
-            np.array([x[metric_2_idx[metric_name]] for metric_name in metrics.keys()]) + np.array([delta[rank] for _, rank in metrics.values()]),
+            np.array([x[_METRIC_2_IDX[metric_name]] for metric_name in metrics.keys()]) + np.array([delta[rank] for _, rank in metrics.values()]),
             [metric_value for metric_value, _ in metrics.values()],
             width=bar_width,
-            label=model,
+            label=model[:-2],
         )
     
     # plot LightGCN performance
     ax.bar(
         x + delta[n],
-        [model_2_metrics['degree_norm|direct|mean'][name] for name in metric_names],
+        [model_2_metrics['degree_norm|direct|mean'][name] for name in _METRIC_NAMES],
         width=bar_width,
         label='LightGCN'
     )
     
-    #plot best baseline performance
+    # plot best baseline performance
     for best_baseline, metrics in best_baseline_2_metrics.items():
         ax.bar(
-            np.array([x[metric_2_idx[metric_name]] for metric_name in metrics.keys()]) + delta[n + 1],
+            np.array([x[_METRIC_2_IDX[metric_name]] for metric_name in metrics.keys()]) + delta[n + 1],
             [metric_value for metric_value in metrics.values()],
             width=bar_width,
-            label=best_baseline
+            label=best_baseline if best_baseline != 'KNNBa' else 'KNN'
         )
     
     ax.set_ylabel('Metric Value')
     ax.set_title(f'Rank-based Metrics of Top Performing GNNs\nbenchmarked by LightGCN and the Best Baseline Model\n(Epochs = {unique_epochs})')
-    ax.set_xticks(x, metric_display_names)
+    ax.set_xticks(x, _METRIC_DISPLAY_NAMES)
     ax.legend(loc='upper right', bbox_to_anchor=(1.12, 1.0))
     
     if save_path is not None:
         fig.savefig(save_path, dpi=300)
 
+
+def get_architectures(component: str = 'info_updater', seed: int = 0):
```

Comment:
Maybe try another seed for the final node rep plot. See if performance can be improved